.github/workflows/ci_cd.yml

```diff
@@ -74,12 +74,19 @@ jobs:
       - name: Install pyfluent-parametric
         run: make install
 
+      - name: Generate token for bot
+        id: get_workflow_token
+        uses: peter-murray/workflow-application-token-action@v2
+        with:
+          application_id: ${{ secrets.BOT_APPLICATION_ID }}
+          application_private_key: ${{ secrets.BOT_APPLICATION_PRIVATE_KEY }}
+
       - name: Login to GitHub Container Registry
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           registry: ghcr.io
-          username: ${{ secrets.GH_USERNAME }}
-          password: ${{ secrets.REPO_DOWNLOAD_PAT }}
+          username: ansys-bot
+          password: ${{ steps.get_workflow_token.outputs.token }}
```

Comment:
@RobPasMue Could you please help to get this setting correctly? I'm trying to avoid using personal PATs which also don't work for dependabots. In https://github.com/ansys/pyfluent, where the Fluent docker image is linked as a package, setting `${{ secrets.GITHUB_TOKEN }}` as password can download the image. I tried `${{ secrets.GITHUB_TOKEN }}` here before, that didn't work (I guessed that's because GITHUB_TOKEN is specific to a github repo). The generated token above doesn't work either. It can login to ghcr.io but cannot download the image.

Reply:
HI @mkundu1! Let me have a look :)